src/transactions/TransactionFrame.cpp

```diff
@@ -292,9 +292,12 @@ TransactionFrame::commonValid(SignatureChecker& signatureChecker,
         return false;
     }
 
+    auto balanceAfter = (applying && (app.getLedgerManager().getCurrentLedgerVersion() > 8))
```

Comment:
this can be done after fix-cache-issues PR is merged, now it would be just a waste of time (as it would have to be redone), so I'm sticking to simple ifs